### DISCUSSION TITLE
USRP & O-RU dependency clarification

### DISCUSSION THREAD

**Discussion by Manimaranchandran** [srsran/srsRAN_Project#344] [General Help]
Hi Team, Hope you are doing well !
I would like to have detailed understanding on below queries. Please support to clarify the same.
We have currently built 5G ORAN setup using following configuration in single VM,
srsRAN4G <--(ZMQ)--> gNB (srsRAN) <---> Open5GS 
 **Server: DELLR630 |  CPU: 4 Cores |  RAM: 8 GB | OS: Ubuntu 22.04.3 LTS | gNB : srsRAN gNB software | RU,UE : srs ZMQ Virtual RU+UE |  CORE : Open5GS**

Q1 --> With the above setup, verified UE attach & detach. UL & DL Ping verification.
Q2 --> Would like to know what are the other possible tests can be done with this setup?
Q3 --> Having core + RAN in same VM is acceptable? There is a hint to run iperf3 on server side, having single VM configuration where we should run the iperf3 for the server?

Q4 --> I could see 7.2 Guide suggested to use O-RU from Foxconn & BNTL. Is there any other way we can run this test with USRP (ORAN complaint device) if available/possible?

Q5--> We have Sercomm RU which supports for Split2. Can we still run the test with srsgNB + Sercomm split2 RU. Does it work smoothly?

Q6 --> If we use USRP device that will not be a ORAN compliant test is my understanding. Please correct me if understanding is wrong.

Kindly help us to move forward.

**Comment by pgawlowicz** [comment]
Hi, answers below:
Q1 -> great!
Q2 -> Please check our tutorials for a general overview of the setup functionality. Please be more specific about what you would like to achieve.
Q3 -> Since with ZMQ the setup runs in emulation mode, it is acceptable. iperf3 server has to run on the UE or outside the 5G network.
Q4 -> We do have an example with USRP b210, see here: https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html 
Q5 -> We never tried with Sercomm RU. Does it support split 2 or 7.2?
Q6 -> Not sure if I understand you correctly. USRP is an RF front-end and it will transmit/receive 5G signal, but internally we use UHD driver to transmit/receive samples. In the case of the commercial RUs, we use the OFH interface.
